src/jinja2/nativetypes.py

```diff
@@ -37,6 +37,31 @@ def native_concat(nodes):
         return raw
 
 
+async def native_concat_async(async_gen):
+    nodes = []
+
+    async def collect():
+        async for event in async_gen:
+            nodes.append(event)
+
+    await collect()
```

Comment:
This extra level doesn't seem necessary, can the `async for` be used directly? If not, this needs a comment explaining why.